### DISCUSSION TITLE
Write files to disk based on "last access to date" order

### DISCUSSION THREAD

**Discussion by jamesvclements** [hadynz/obsidian-kindle-plugin#217] [Ideas]
It'd be great if the files were written to match the `last access date` order, so that when you sort your vault by `"Modified time (new to old)"` the books you've accessed most recently are at the top, just like the notes you've accessed most recently are at the top.
The current order seems to be the opposite. 

![How to Win Friends and Influence People (Dale Carnegie) - main - Obsidian v1 0 3-Wednesday-November-09-2022-11 22 16AM@2x](https://user-images.githubusercontent.com/20052710/200922072-5aa682c4-52c7-477a-8e36-a7c0b3619e1c.png)

**Comment by hadynz** [comment]
@jamesvclements that should already be the case right now.

If you are syncing from Amazon Cloud, the plugin looks for any books that have been accessed since it last synced and attempts to resync the book's highlights again. After such an attempt, it updates the front matter at the top of the file with the new last synced date that it just retrieved. That update is obviously a modification to the physical file and hence should show up when you order files by date.

![image](https://user-images.githubusercontent.com/315585/200934047-e0813cc6-cc1b-4772-a87d-5cd4ba6f716a.png)

**Reply by jamesvclements** [reply]
@hadynz ah okay. I see that working for updated books, but on the very first sync it seemed to be in the opposite order

**Reply by hadynz** [reply]
That's a fair point.

On the very first sync, the plugin iterates through your list of books as they appear on the online Kindle Reader, with not deliberate sorting. This is a change that I can make, however, I don't think this is an ideal solution for a couple of reasons:

* If the plugin needs to be tuned for performance in the future, I might make it download books in parallel. When that happens then there is no guarantee on order
* If someone backs up their vault, and restores it on a new device, the creation/modified date changes to the restoration date

Have you considered using the [Dataview][1] plugin and creating a view that lists all your books using the `lastAnnodatedDate` value in the frontmatter?

[1]: https://github.com/blacksmithgu/obsidian-dataview
